frontend/src/scenes/max/maxContextLogic.ts

```diff
@@ -112,6 +139,28 @@ export const maxContextLogic = kea<maxContextLogicType>([
                 resetContext: () => [],
             },
         ],
+        contextEvents: [
+            [] as MaxEventContext[],
+            {
+                addOrUpdateContextEvent: (state: MaxEventContext[], { event }: { event: EventDefinition }) =>
+                    state.filter((stateEvent) => stateEvent.id !== event.id).concat(eventToMaxContext(event)),
+                removeContextEvent: (state: MaxEventContext[], { id }: { id: string | number }) => {
+                    return state.filter((event) => event.id !== id)
+                },
+                resetContext: () => [],
+            },
+        ],
+        contextActions: [
+            [] as MaxActionContext[],
+            {
+                addOrUpdateContextAction: (state: MaxActionContext[], { action }: { action: ActionType }) =>
+                    state.filter((stateAction) => stateAction.id !== action.id).concat(actionToMaxContext(action)),
+                removeContextAction: (state: MaxActionContext[], { id }: { id: string | number }) => {
+                    return state.filter((action) => action.id !== id)
+                },
+                resetContext: () => [],
+            },
+        ],
```

Comment:
This is becoming boilerplate-heavy. Did you consider as common `context: (MaxEventContext | MaxDashboardContext | etc.)[]` reducer?